src/main/java/com/synopsys/integration/jira/common/cloud/rest/service/IssueService.java

```diff
@@ -36,13 +48,42 @@
     public static final String API_PATH_TRANSITIONS_SUFFIX = "transitions";
     public static final String API_PATH_COMMENTS_SUFFIX = "comment";
 
-    private JiraCloudService jiraCloudService;
+    private final JiraCloudService jiraCloudService;
+    private final UserSearchService userSearchService;
+    private final ProjectService projectService;
+    private final IssueTypeService issueTypeService;
 
-    public IssueService(final JiraCloudService jiraCloudService) {
+    public IssueService(final JiraCloudService jiraCloudService, final UserSearchService userSearchService, final ProjectService projectService, final IssueTypeService issueTypeService) {
         this.jiraCloudService = jiraCloudService;
+        this.userSearchService = userSearchService;
+        this.projectService = projectService;
+        this.issueTypeService = issueTypeService;
+    }
+
+    public IssueResponseModel createIssue(IssueCreationRequestModel requestModel) throws IntegrationException {
+        IssueRequestModelFieldsBuilder fieldsBuilder = new IssueRequestModelFieldsBuilder();
+        fieldsBuilder.copyFields(requestModel.getFieldsBuilder());
+        Optional<IssueTypeResponseModel> bugIssueType = issueTypeService.getAllIssueTypes().stream()
+                                                            .filter(issueType -> issueType.getName().equalsIgnoreCase(requestModel.getIssueTypeName()))
+                                                            .findFirst();
+        Optional<UserDetailsComponent> userDetails = userSearchService.findUser(requestModel.getReporterEmail()).stream()
+                                                         .findFirst();
+
+        PageOfProjectsResponseModel projects = projectService.getProjectsByName(requestModel.getProjectName());
+
+        Optional<ProjectComponent> foundProject = projects.getProjects().stream()
+                                                      .findFirst();
+
+        bugIssueType.ifPresent(issueType -> fieldsBuilder.setIssueType(issueType.getId()));
+        userDetails.ifPresent(user -> fieldsBuilder.setReporter(user.getAccountId()));
+        foundProject.ifPresent(project -> fieldsBuilder.setProject(project.getId()));
```

Comment:
Do we prefer collapsing all of these optional chain methods into the original calls starting on line 66? I don't see any other references besides this ifPresent check.

Reply:
Are theses fields required to created an issue? If they are shouldn't we have an orElseThrow(...) as part of the optional use?

Reply:
If that's the case, then we should leave it as is and add orElseThrow to the lines above